### DISCUSSION TITLE
How to setup oc-bibtex-actions with doom emacs?

### DISCUSSION THREAD

**Discussion by anaxonda** [emacs-citar/citar#277] [Q&A] [answered]
Hello,

I'm unable to set the org-cite-global-bibliography variable. I am using the  recommended org-cite settings from the Read Me file:

;; Set bibliography paths so they are the same.
(defvar my/bibs '("~/bib/references.bib"))

(use-package oc-bibtex-actions
  :bind (("C-c b" . org-cite-insert)
         ("M-o" . org-open-at-point)
         :map minibuffer-local-map
         ("M-b" . bibtex-actions-insert-preset))
  :after (embark oc)
  :config
  (setq bibtex-actions-bibliography my/bibs
        org-cite-global-bibliography my/bibs))

;; Use consult-completing-read for enhanced interface.
(advice-add #'completing-read-multiple :override #'consult-completing-read-multiple)

I am able to set a bib file locally in buffer with #+bibliography, which allows me to cite with C-c b (without setting locally C-c b just shows 'nil').
I have bibtex-completion bibliography variables set. If I use bibtex-actions-insert-citation I am able to see all my references, however they are formatted the old org-ref way of cite:KEY upon insertion.
I'm using Doom (I also have the problem referenced here with embark-act:https://github.com/bdarcus/bibtex-actions/discussions/270) .

**Comment by bdarcus** [comment]
Two things:

1. for org-cite, you should only use `org-cite-insert`; not `bibtex-actions-insert-citation`
2. have you tried the biblio module branch with the `vertico` module? https://github.com/hlissner/doom-emacs/pull/5290

**Reply by anaxonda** [reply]
Can you advise me on how to try the biblio module correctly?


I have the biblio module uncommented in init.el, and in my packages.el I have 
`(package! biblio  :recipe (:host github :repo "bdarcus/doom-emacs" :branch "biblio-org-cite"))`

This is evidently not working right. Not sure if I need to unpin something as well?
Thanks!

**Reply by bdarcus** [reply]
Probably the easiest until it's merged is to just copy the init and config files from the PR to your local module directory.

```console
~/.doom.d/modules/tools/biblio/
```

That will then override the core module.

**Reply by anaxonda** [reply]
Thanks. I tried that, but now emacs will not restart without error.


`Error in private config: config.el, (file-missing Cannot open load file No such file or directory biblio)`

Biblio directory exists with the current PR in `~/.doom.d/modules/tools/biblio/`.

**Reply by bdarcus** [reply]
I'm not clear what you did, but you don't need to change you config or init files at all; just place the PR modules files in the correct folder wherever your doom config is.

Here's what my doom dir structure looks like:

```console
❯ tree ~/.config/doom/modules/
/home/bruce/.config/doom/modules/
└── tools
    ├── biblio
    │   ├── config.el
    │   └── packages.el
```

**Reply by anaxonda** [reply]
Well, I finally had success, and everything seems to be working. 

I kept getting various errors. In the end copying the PR module files over to the doom dir, commenting the biblio module in init.el and running doom sync, then uncommenting and running doom sync again seemed to do it.

**Reply by bdarcus** [reply]
Great!

And to be clear, you get the bibtex-actions UI when running `org-cite-insert`?

**Reply by anaxonda** [reply]
Sorry for the late reply, yes, I do get the bibtex-actions UI now. Thanks!

**Comment by bdarcus** [comment, marked as answer]
For this Doom user, the solution was using the (currently unmerged Doom PR 5290).

https://github.com/bdarcus/bibtex-actions/discussions/277#discussioncomment-1293708

I reworded the question accordingly.

**Comment by bdarcus** [comment]
And that PR has since been merged.
